libp2p/network/connection/swarm_connection.py

```diff
@@ -45,43 +51,32 @@ def __init__(self, muxed_conn: IMuxedConn, swarm: "Swarm") -> None:
             await stream.reset()
         # Force context switch for stream handlers to process the stream reset event we just emit
         # before we cancel the stream handler tasks.
-        await asyncio.sleep(0.1)
+        await trio.sleep(0.1)
 
-        for task in self._tasks:
-            task.cancel()
-            try:
-                await task
-            except asyncio.CancelledError:
-                pass
-        # Schedule `self._notify_disconnected` to make it execute after `close` is finished.
         self._notify_disconnected()
 
     async def _handle_new_streams(self) -> None:
+        self.event_started.set()
         while True:
             try:
                 stream = await self.muxed_conn.accept_stream()
             except MuxedConnUnavailable:
-                # If there is anything wrong in the MuxedConn,
-                # we should break the loop and close the connection.
                 break
             # Asynchronously handle the accepted stream, to avoid blocking the next stream.
-            await self.run_task(self._handle_muxed_stream(stream))
+            self.swarm.manager.run_task(self._handle_muxed_stream, stream)
 
         await self.close()
 
-    async def _call_stream_handler(self, net_stream: NetStream) -> None:
-        try:
-            await self.swarm.common_stream_handler(net_stream)
-        # TODO: More exact exceptions
-        except Exception:
-            # TODO: Emit logs.
-            # TODO: Clean up and remove the stream from SwarmConn if there is anything wrong.
-            self.remove_stream(net_stream)
-
     async def _handle_muxed_stream(self, muxed_stream: IMuxedStream) -> None:
         net_stream = self._add_stream(muxed_stream)
         if self.swarm.common_stream_handler is not None:
-            await self.run_task(self._call_stream_handler(net_stream))
+            try:
+                await self.swarm.common_stream_handler(net_stream)
+            # TODO: More exact exceptions
+            except Exception:
```

Comment:
Are you intentionally silencing this exception?  This looks like you may want to just do a `try/finally` and allow the exception to bubble.

Reply:
I intended to catch exceptions raised in the `common_stream_handler` to avoid the whole `Swarm` broken. After second thoughts, it seems not what we should take care of. And it seems I forgot to do `remove_stream` for the success case. Nice catch!